v2/env_metadata.go

```diff
@@ -0,0 +1,45 @@
+package bugsnag
+
+import (
+	"fmt"
+	"strings"
+)
+
+const metadataPrefix string = "BUGSNAG_METADATA_"
+const metadataPrefixLen int = len(metadataPrefix)
+
+type envMetadata struct {
+	tab   string
+	key   string
+	value string
+}
+
+func loadEnvMetadata(environ []string) []envMetadata {
+	metadata := make([]envMetadata, 0)
+	for _, value := range environ {
+		key, value, err := parseEnvironmentPair(value)
+		if err != nil {
+			continue
+		}
+		if keypath, err := parseMetadataKeypath(key); err == nil {
+			tab, key := splitTabKeyValues(keypath)
+			metadata = append(metadata, envMetadata{tab, key, value})
+		}
+	}
+	return metadata
+}
+
+func splitTabKeyValues(keypath string) (string, string) {
+	key_components := strings.SplitN(keypath, "_", 2)
+	if len(key_components) > 1 {
+		return key_components[0], key_components[1]
+	}
+	return "custom", keypath
```

Comment:
Might it be an idea to extract the "custom" default key name as a const to make it self-documenting?